### DISCUSSION TITLE
Can I update a vsm built-in shape or at least the anchor of it ?

### DISCUSSION THREAD

**Discussion by malexisw** [clientIO/joint#2404] [Q&A] [answered]
### Introduction

Hello, 
My issue is that, when i create an arrow between 2 shapes and change it to a bigger one, the head of that same arrow goes to far in the shape. 

The big arrow in the example is vsm.VSMMaterialFlow

### Steps to reproduce

<img width="359" alt="Capture d’écran 2023-11-23 à 14 56 25" src="https://github.com/clientIO/joint/assets/67229418/02304084-b832-4f5c-9136-782afff44ca2">
<img width="359" alt="Capture d’écran 2023-11-23 à 14 56 33" src="https://github.com/clientIO/joint/assets/67229418/a7ef6d15-dd4a-4a08-8c4b-0d7bf6c1d432">

### Restrictions & Constraints

Do i need to make a custom arrow and not take the one from the library to be able to change it or update the anchor of it or maybe make the shape bigger each time i change to a bigger arrow ?

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS, JointJS+

**Comment by kumilingus** [comment, marked as answer]
It might be best to modify the connection point offset based on the link type, e.g. `VSMMaterialFlow`.

```js
paper.options.defaultConnectionPoint = function(line, view, magnet, opt, type, linkView) {
    const newOpt = { ...opt };
    if (linkView.model.get('type') !== 'VSMMaterialFlow') {
        newOpt.offset = (type === 'source') ? 10 : 20;
    }
    return joint.connectionPoints.boundary.call(this, line, view, magnet, newOpt, type, linkView);
},
```
